File: Algorithms/718.Maximum-Length-of-Repeated-Subarray/maximum_length_of_repeated_subarray.go
func findLength(A []int, B []int) int {
	lenA, lenB := len(A), len(B)
	dp := make([][]int, lenA)
	for i := range dp {
		dp[i] = make([]int, lenB)
	}

	res := 0
	for i := 0; i < lenA; i++ {
		for j := 0; j < lenB; j++ {
			if A[i] == B[j] {
				if i == 0 || j == 0 {
					dp[i][j] = 1
				} else {
					dp[i][j] = dp[i-1][j-1] + 1
				}
				res = max(res, dp[i][j])
			}
		}
	}
	return res
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

File: Algorithms/718.Maximum-Length-of-Repeated-Subarray/maximum_length_of_repeated_subarray.py
# -*- coding: utf-8 -*-

class Solution:
    def findLength(self, A, B):
        """
        :type A: List[int]
        :type B: List[int]
        :rtype: int
        """
        lenA = len(A)
        lenB = len(B)
        dp = [[0 for j in range(lenB)] for i in range(lenA)]
        res = 0
        for i in range(lenA):
            for j in range(lenB):
                if A[i] == B[j]:
                    if i == 0 or j == 0:
                        dp[i][j] = 1
                    else:
                        dp[i][j] = dp[i - 1][j - 1] + 1
                    res = max(res, dp[i][j])
        return res
